microsoft/testsuites/core/storage.py

```diff
@@ -242,6 +251,244 @@ def verify_os_partition_identifier(self, log: Logger, node: RemoteNode) -> None:
                 "'PARTUUID={os_partition_info.part_uuid} /' should be present in fstab"
             )
 
+    @TestCaseMetadata(
+        description="""
+        This test case will verify that the standard hdd data disks disks can
+        be added serially while the vm is running.
+        Steps:
+        1. Get maximum number of data disk for the current vm_size.
+        2. Get the number of data disks already added to the vm.
+        3. Serially add and remove the data disks and verify that the added
+        disks are present in the vm.
+        """,
+        requirement=simple_requirement(disk=DiskStandardHDDLRS()),
+    )
+    def hot_add_disk_serial(self, log: Logger, node: Node) -> None:
+        self._hot_add_disk_serial(
+            log, node, DiskType.StandardHDDLRS, self.DEFAULT_DISK_SIZE_IN_GB
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case will verify that the standard ssd data disks disks can
+        be added serially while the vm is running. The test steps are same as
+        `hot_add_disk_serial`.
+        """,
+        requirement=simple_requirement(disk=DiskStandardSSDLRS()),
+    )
+    def hot_add_disk_serial_standard_ssd(self, log: Logger, node: Node) -> None:
+        self._hot_add_disk_serial(
+            log, node, DiskType.StandardSSDLRS, self.DEFAULT_DISK_SIZE_IN_GB
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case will verify that the premium ssd data disks disks can
+        be added serially while the vm is running. The test steps are same as
+        `hot_add_disk_serial`.
+        """,
+        requirement=simple_requirement(disk=DiskPremiumSSDLRS()),
+    )
+    def hot_add_disk_serial_premium_ssd(self, log: Logger, node: Node) -> None:
+        self._hot_add_disk_serial(
+            log, node, DiskType.PremiumSSDLRS, self.DEFAULT_DISK_SIZE_IN_GB
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case will verify that the standard HDD data disks can
+        be added in parallel while the vm is running.
+        Steps:
+        1. Get maximum number of data disk for the current vm_size.
+        2. Get the number of data disks already added to the vm.
+        3. Add maximum number of data disks to the VM in parallel.
+        4. Verify that the disks are added are available in the OS.
+        5. Remove the disks from the vm in parallel.
+        6. Verify that the disks are removed from the OS.
+        """,
+        requirement=simple_requirement(disk=DiskStandardHDDLRS()),
+    )
+    def hot_add_disk_parallel(self, log: Logger, node: Node) -> None:
```

Comment:
What does "parallel" mean here?

Reply:
It's basically that all the disks are added in one go rather than one by one

Reply:
I will update the steps to make it more clear